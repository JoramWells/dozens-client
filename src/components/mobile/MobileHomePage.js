import { ThumbUpIcon } from "@heroicons/react/solid";
import React from "react";
import { Helmet } from "react-helmet";

export default function MobileHomePage() {
  return (
    <div className="mobile__body">
      <Helmet>
        <meta charSet="utf-8" />
        <title>Dozens Kenya</title>
      </Helmet>
      <main className="p-4">
        <div className="flex flex-col space-y-8">
          <figure className="shadow-lg">
            <img
              src="https://images.unsplash.com/flagged/photo-1556637640-2c80d3201be8?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=750&q=80"
              style={{ width: "25rem", height: "200px" }}
              className="rounded-t-md"
              alt="shoe1"
            />
            <div className="p-2">
              <blockquote>product name</blockquote>
              <figcaption>
                <div>
                  <p className="font-semibold text-gray-600">3,456/=</p>
                </div>
                <div className="line-through text-sm text-gray-500">
                  12,344/=
                </div>
                <div>
                  <ThumbUpIcon className="h-5 text-gray-500" />
                </div>
              </figcaption>
            </div>
          </figure>
          <figure className="shadow-lg">
            <img
              src="https://images.unsplash.com/photo-1542291026-7eec264c27ff?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=750&q=80"
              style={{ width: "25rem", height: "200px" }}
              className="rounded-t-md"
              alt="shoe2"
            />
            <div className="p-2">
              <blockquote>product name</blockquote>
              <figcaption>
                <div>
                  <p className="font-semibold text-gray-600">3,456/=</p>
                </div>
                <div className="line-through text-sm text-gray-500">
                  12,344/=
                </div>
                <div>
                  <ThumbUpIcon className="h-5 text-gray-500" />
                </div>
              </figcaption>
            </div>
          </figure>
          <figure className="border-1">
            <img
              src="https://images.unsplash.com/photo-1460353581641-37baddab0fa2?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=751&q=80"
              style={{ width: "25rem", height: "200px" }}
              className="rounded-t-md"
              alt="shoe3"
            />
            <div className="p-2">
              <blockquote>product name</blockquote>
              <figcaption>
                <div>
                  <p className="font-semibold text-gray-600">3,456/=</p>
                </div>
                <div className="line-through text-sm text-gray-500">
                  12,344/=
                </div>
                <div>
                  <ThumbUpIcon className="h-5 text-gray-500" />
                </div>
              </figcaption>
            </div>
          </figure>
        </div>
      </main>
    </div>
  );
}
