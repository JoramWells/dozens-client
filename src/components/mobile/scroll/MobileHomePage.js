import { ThumbUpIcon } from "@heroicons/react/solid";
import React from "react";
import { Helmet } from "react-helmet";

export default function MobileHomePage({product}) {
  const {id,alt,productName,price,discount,likes} = product
  return (
    <div className="mobile__body" data-testid={`product-${id}`}>
      <Helmet>
        <title>HomePage | Dozens Kenya</title>
      </Helmet>

      <div className="p-4">
        <div className="flex flex-col content-center items-center space-y-8">
          <figure style={{ width: "18rem", border: "1px solid #E8E8E8 " }} className="rounded-md">
            <img
              src="https://images.unsplash.com/photo-1460353581641-37baddab0fa2?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=751&q=80"
              style={{ width: "18rem", height: "170px" }}
              alt={alt}
            />
            <div className="p-2">
              <blockquote>{productName}</blockquote>
              <p className="text-gray-600">
              Airforce Maziwa

              </p>
              <figcaption>
                <div>
                  <p className="font-semibold text-gray-600">{price}</p>
                  <p className="text-lg font-semibold text-yellow-400">
                  Kshs 2,213/=

                  </p>
                </div>
                <div className="line-through text-sm text-gray-300">
                  {discount}
                  Kshs 4,321/=
                </div>
                <hr className="my-2" style={{color:"#A9A9A9"}}/>
                <div className="flex flex-row space-x-2 items-center content-center">
                  {likes}
                  <p className="text-sm text-gray-400">
                  231
                  </p>
                  <ThumbUpIcon className="h-5 text-gray-500" />
                </div>
              </figcaption>
            </div>
          </figure>
        </div>
      </div>
    </div>
  );
}
