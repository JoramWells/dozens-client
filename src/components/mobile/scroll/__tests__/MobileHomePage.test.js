import { render, cleanup, screen } from "@testing-library/react";
import MobileHomePage from "../MobileHomePage";
import renderer from 'react-test-renderer'

afterEach(()=>{
    cleanup()
})
test("should render infinite scroll", () => {
  const product = {
    id: 1,
    src: "https://images.unsplash.com/flagged/photo-1556637640-2c80d3201be8?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=750&q=80",
    alt: "img1",
    productName: "img1",
    price: "1234/=",
    discount: "4,322/=",
    likes: "45",
  };
  render(<MobileHomePage product={product} />);
  const productElement = screen.getByTestId("product-1")
  expect(productElement).toBeInTheDocument()
  expect(productElement).toHaveTextContent("img1")
});
test("should have discount price", () => {
  const product = {
    id: 1,
    src: "https://images.unsplash.com/flagged/photo-1556637640-2c80d3201be8?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=750&q=80",
    alt: "img1",
    productName: "air force",
    price: "1234/=",
    discount: "4,322/=",
    likes: "45",
  };
  render(<MobileHomePage product={product} />);
  const productElement = screen.getByTestId("product-1")
  expect(productElement).toBeInTheDocument()
  expect(productElement).toHaveTextContent("air force")
  // expect(productElement).toContainHTML("<strike>")

});

test("matches snapshot",()=>{
  const product = {
    id: 1,
    src: "https://images.unsplash.com/flagged/photo-1556637640-2c80d3201be8?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=750&q=80",
    alt: "img1",
    productName: "air force",
    price: "1234/=",
    discount: "4,322/=",
    likes: "45",
  };
  const tree = renderer.create(<MobileHomePage product={product} />).toJSON()
  expect(tree).toMatchSnapshot();

})

