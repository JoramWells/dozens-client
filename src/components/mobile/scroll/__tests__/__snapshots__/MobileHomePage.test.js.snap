// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="mobile__body"
  data-testid="product-1"
>
  <div
    className="p-4"
  >
    <div
      className="flex flex-col content-center items-center space-y-8"
    >
      <figure
        style={
          Object {
            "border": "1px solid whitesmoke",
            "width": "18rem",
          }
        }
      >
        <img
          alt="img1"
          src="https://images.unsplash.com/flagged/photo-1556637640-2c80d3201be8?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=750&q=80"
          style={
            Object {
              "height": "170px",
              "width": "18rem",
            }
          }
        />
        <div
          className="p-2"
        >
          <blockquote>
            air force
          </blockquote>
          <figcaption>
            <div>
              <p
                className="font-semibold text-gray-600"
              >
                1234/=
              </p>
            </div>
            <div
              className="line-through text-sm text-gray-500"
            >
              4,322/=
            </div>
            <div
              className="flex flex-row space-x-2"
            >
              45
              <svg
                className="h-5 text-gray-500"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2 10.5a1.5 1.5 0 113 0v6a1.5 1.5 0 01-3 0v-6zM6 10.333v5.43a2 2 0 001.106 1.79l.05.025A4 4 0 008.943 18h5.416a2 2 0 001.962-1.608l1.2-6A2 2 0 0015.56 8H12V4a2 2 0 00-2-2 1 1 0 00-1 1v.667a4 4 0 01-.8 2.4L6.8 7.933a4 4 0 00-.8 2.4z"
                />
              </svg>
            </div>
          </figcaption>
        </div>
      </figure>
    </div>
  </div>
</div>
`;
